types/src/v0/impls/header.rs

```diff
@@ -1101,9 +1101,9 @@ impl BlockHeader<SeqTypes> for Header {
         })
     }
 
-    fn auth_root(&self) -> anyhow::Result<Option<[u8; 32]>> {
+    fn auth_root(&self) -> anyhow::Result<FixedBytes<32>> {
         match self {
-            Header::V1(_) | Header::V2(_) | Header::V3(_) => Ok(None),
+            Header::V1(_) | Header::V2(_) | Header::V3(_) => Ok(FixedBytes::from([0u8; 32])),
```

Comment:
Nit: I do think using Option is better if the field is not present for versions <= V3, instead of using zeros

Reply:
How about putting the `Option` here but no `Result`? It's hard to imagine a failure case.